node/network/availability-recovery/src/lib.rs

```diff
@@ -686,6 +687,38 @@ impl Future for RecoveryHandle {
 	}
 }
 
+/// Cached result of an availability recovery operation.
+#[derive(Debug, Clone)]
+enum CachedResult {
+	/// Availability was successfully retrieved before.
+	Success(AvailableData),
+	/// Availability was successfully retrieved before, but was found to be invalid.
+	Invalid,
+}
```

Comment:
Maybe we can work with names. Like use `Invalid` and `Valid` for variants. Since that will make it related to validity we can name the enum `CachedValidity` as well? This will solve unfortunate naming of `CachedResult::into_result` which sounds a bit tautology.